### PR TITLE
Add BatchNorm layer to Digit Recognizer

### PR DESCRIPTION
**Pull request by Manthan-R-Sheth**
This PR creates a new model for Digit Recognition task of Kaggle including the batch norm layer.

### CODE REVIEW

File: Kaggle/DigitRecognizer/src/DigitRecognizerBatchNorm.cpp [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            // up to reaching maximum of iterations.
+            1e-8,
+            // Shuffle. If optimizer should take random data points from the dataset at
+            // each iteration.
+            true,
+            // Adam update policy.
+            AdamUpdate(1e-8, 0.9, 0.999));
+
+    // Cycles for monitoring the process of a solution.
+    for (int i = 0; i <= CYCLES; i++) {
```
Comment (zoq) [inline Kaggle/DigitRecognizer/src/DigitRecognizerBatchNorm.cpp]: I think we should try and use a similar style as we use for the main repo (put the opening ``{`` on a new line, indentation 2 spaces, tab twice if we make a new line).

**Comment by Manthan-R-Sheth** [comment]
Sure @zoq and @rcurtin , I will look into this by tonight and update the PR.
I had tested it on this dataset already when working on implementing it [here](https://github.com/mlpack/mlpack/pull/1275) and the results looked quite promising.

**Comment by Manthan-R-Sheth** [comment]
@rcurtin @zoq 
I have done the changes, could you review it once.

File: Kaggle/DigitRecognizer/CMakeLists.txt [hunk 2]
```diff
@@ -21,6 +21,6 @@ target_link_libraries(DigitRecognizer
   ${ARMADILLO_LIBRARIES})
 
 add_custom_command(TARGET DigitRecognizer POST_BUILD
-  COMMAND ${CMAKE_COMMAND} -E copy_directory 
-    ${CMAKE_CURRENT_SOURCE_DIR}/data/ 
+  COMMAND ${CMAKE_COMMAND} -E copy_directory
+    kaggle-digit-recognizer_SOURCE_DIR/data/
```
Comment (rcurtin) [inline Kaggle/DigitRecognizer/CMakeLists.txt]: Should this be a CMake variable, like `${kaggle-digit-recognizer_SOURCE_DIR}`?  I am not sure that works, by the way---I haven't tested it.  You could also do `${CMAKE_CURRENT_SOURCE_DIR}/../DigitRecognizer/data/`, which might be better.

File: Kaggle/DigitRecognizerBatchNorm/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 2.6)
+project(kaggle-digit-recognizer)
```
Comment (rcurtin) [inline Kaggle/DigitRecognizerBatchNorm/CMakeLists.txt]: Probably this needs a different name so as not to collide with the existing `kaggle-digit-recognizer` project.

File: Kaggle/DigitRecognizerBatchNorm/src/DigitRecognizerBatchNorm.cpp [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+using namespace mlpack::ann;
+using namespace mlpack::optimization;
+
+using namespace arma;
+using namespace std;
+
+int main() {
+  // Dataset is randomly split into training
+  // and validation parts with following ratio.
+  constexpr double RATIO = 0.9;
```
Comment (rcurtin) [inline Kaggle/DigitRecognizerBatchNorm/src/DigitRecognizerBatchNorm.cpp]: Should this be `0.1`?  See #8.

**Comment by zoq** [comment]
This looks good to me, would be great if you could post some results here, I guess after that this is ready to be merged. Thanks for looking into all the issues.

**Comment by Manthan-R-Sheth** [comment]
@zoq @rcurtin 
Without BatchNorm,
withoutbatchnorm
  
With BatchNorm,
batchnorm

**Comment by rcurtin** [comment]
I think it was just forgotten---there's no AppVeyor config, so this PR wouldn't have broken anything.  I'll go ahead and merge it now.  Thanks @akhandait for pointing it out. :)
